jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java

```diff
@@ -84,11 +84,11 @@ public JmxTransModule(@Nonnull JmxTransConfiguration configuration) {
 	@Override
 	protected void configure() {
 		bind(new TypeLiteral<GenericKeyedObjectPool<InetSocketAddress, Socket>>(){})
-				.toInstance(getObjectPool(new SocketFactory(), SocketFactory.class.getSimpleName()));
+				.toInstance(getObjectPool(new SocketFactory(), SocketFactory.class.getSimpleName(), 0));
 		bind(new TypeLiteral<GenericKeyedObjectPool<SocketAddress, DatagramSocket>>(){})
-				.toInstance(getObjectPool(new DatagramSocketFactory(), DatagramSocketFactory.class.getSimpleName()));
+				.toInstance(getObjectPool(new DatagramSocketFactory(), DatagramSocketFactory.class.getSimpleName(), 0));
 		bind(new TypeLiteral<KeyedObjectPool<JmxConnectionProvider, JMXConnection>>(){}).annotatedWith(Names.named("mbeanPool"))
-				.toInstance(getObjectPool(new MBeanServerConnectionFactory(), MBeanServerConnectionFactory.class.getSimpleName()));
+				.toInstance(getObjectPool(new MBeanServerConnectionFactory(), MBeanServerConnectionFactory.class.getSimpleName(), 20000));
```

Comment:
It would be great to expose this as a cli argument (but not absolutely required). We need to introduce a config file instead of passing all options via cli, the current argument processing is messy and overly complex.

Reply:
It's good idea to make it parameter. I wrote your thoughts about config file at https://github.com/jmxtrans/jmxtrans/pull/566 and agree with them, but it seems like separate task.

So I want to extract it like cli parameter. Is it okay if I will do another pull request for this?

Reply:
sure, all good!